plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/PodDeploymentPathProvider.java

```diff
@@ -51,18 +51,22 @@
 
 	private static final String DOCKER_IMAGE_DIGEST_IDENTIFIER = "sha256:";
 
-	public String load(IService service, Connection connection) throws CoreException {
-		IProject project = service.getProject();
-		IPod pod = getPod(service, project, connection);
+	public String load(IResource resource, Connection connection) throws CoreException {
+		IProject project = resource.getProject();
+		IPod pod = getPod(resource, project, connection);
 		String imageRef = getImageRef(connection, project, pod);
 		int imageDigestIndex = imageRef.indexOf(DOCKER_IMAGE_DIGEST_IDENTIFIER);
 		if (imageDigestIndex > 0) {
 			String imageDigest = imageRef.substring(imageDigestIndex);
 			IResource imageStreamTag = getImageStreamTag(connection, project, imageRef, imageDigest);
 			return getPodPath(imageStreamTag);
 		} else {
-			
-			return "";
+		    String imageMetaData = getImageMetaData(imageRef, project);
+		    if (imageMetaData != null) {
```

Comment:
the if/else is required for deployment from docker images without sha256 id that is without image stream tags. This is what I used when deploying without a deployment config. And null or "" is handled the same way.